Mathlib/Data/Fin/Basic.lean

```diff
@@ -144,6 +193,19 @@ instance : SubNegMonoid (Fin n.succ) where
 instance : AddGroup (Fin n.succ) where
   add_left_neg := Fin.add_left_neg
 
+instance : Monoid (Fin n.succ) where
```

Comment:
Now that I see it, we could factor out `CommSemigroup (Fin n)` and `AddCommSemigroup (Fin n)` instances (without the succ).